aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsFluentClientDecorator.kt

```diff
@@ -88,33 +112,34 @@ class AwsFluentClientDecorator : RustCodegenDecorator {
     }
 }
 
-private class AwsFluentClientExtensions(private val types: Types) {
-    val clientGenerics = ClientGenerics(
-        connectorDefault = types.dynConnector,
-        middlewareDefault = types.defaultMiddleware,
-        retryDefault = types.smithyClientRetry.member("Standard"),
-        client = types.awsSmithyClient
-    )
-
+private class AwsFluentClientExtensions(types: Types) {
     private val codegenScope = arrayOf(
         "Middleware" to types.defaultMiddleware,
         "retry" to types.smithyClientRetry,
         "DynConnector" to types.dynConnector,
-        "aws_smithy_client" to types.awsSmithyClient
+        "DynMiddleware" to types.dynMiddleware,
+        "SmithyConnector" to types.smithyConnector,
+        "ConnectorError" to types.connectorError,
+        "aws_smithy_client" to types.awsSmithyClient,
+        "aws_types" to types.awsTypes,
     )
 
     fun render(writer: RustWriter) {
-        writer.rustBlockTemplate("impl<C> Client<C, #{Middleware}, #{retry}::Standard>", *codegenScope) {
+        writer.rustBlockTemplate("impl Client", *codegenScope) {
             rustTemplate(
                 """
                 /// Creates a client with the given service config and connector override.
-                pub fn from_conf_conn(conf: crate::Config, conn: C) -> Self {
+                pub fn from_conf_conn<C, E>(conf: crate::Config, conn: C) -> Self
+                where
+                    C: #{SmithyConnector}<Error = E> + Send + 'static,
+                    E: Into<#{ConnectorError}>,
+                {
                     let retry_config = conf.retry_config.as_ref().cloned().unwrap_or_default();
                     let timeout_config = conf.timeout_config.as_ref().cloned().unwrap_or_default();
                     let sleep_impl = conf.sleep_impl.clone();
                     let mut builder = #{aws_smithy_client}::Builder::new()
-                        .connector(conn)
-                        .middleware(#{Middleware}::new());
+                        .connector(#{DynConnector}::new(conn))
+                        .middleware(#{DynMiddleware}::new(#{Middleware}::new()));
```

Comment:
we _could_ consider sticking these types into the client generics for a little `DRY` but not sure if it's worth it